engines/e_ossltest.c

```diff
@@ -13,6 +13,13 @@
  * used for any purpose except testing
  */
 
+/*
+ * SHA low level APIs are deprecated for public use, but still ok for
+ * internal use.  Note, that due to symbols not being exported, only the
+ * #defines can be accessed.
+ */
+#include "internal/deprecated.h"
```

Comment:
Perhaps just define SHA_CBLOCK locally?

Reply:
I strongly disagree with having that defined "locally" in several places

Reply:
That is, unless there's justification, such as values can vary depending on location. I assume that's not the case here.

Reply:
I'm with @levitte here, best keep the definition in one place.  For the test engine, I don't see a problem with accessing the symbols directly (it needs more than just `SHA_CBLOCK` BTW).  The async engine is example code unless I'm mistaken and it should do the right thing.

I could call `EVP_MD_meth_get_input_blocksize(EVP_sha())` or similar to get the value and set that into the created EVP_MD.  This seems like an all around better approach.

Reply:
All good plans seem to fail.  The engines are also directly accessing sizeof(SHA_CTX) and similar.

Reply:
So?  the whole ENGINE API will become deprecated as well...  (I gotta finish up that branch!)

Reply:
This is better left for the ENGINE deprecation -- I can't see us dropping the either the test or async engine code soon.

I updated the comment to include the `sizeof(ctx)` bit.